src/charm.py

```diff
@@ -319,10 +326,41 @@ def _update_ingress(self) -> None:
             ingress_controller = self._get_ingress_controller(namespace)
             ingress_ips = ingress_controller.get_ingress_ips()
             message = f"Ingress IP(s): {', '.join(ingress_ips)}" if ingress_ips else ""
+
+            if self._get_definition_from_relation(relation).is_ingress_relation:
```

Comment:
You already have the `definition` variable available (l314) so you can use it directly